src/HttpServer.h

```diff
@@ -5,48 +5,71 @@ All rights reserved.
 Use of this source code is governed by a BSD-style license that can be
 found in the LICENSE file. See the AUTHORS file for names of contributors.
  */
+#include <boost/array.hpp>
+#include <boost/asio.hpp>
+#include <boost/bind/bind.hpp>
+#include <boost/enable_shared_from_this.hpp>
+#include <boost/shared_ptr.hpp>
+#include <ctime>
 #include <iostream>
+#include <string>
 
 #include "ThreadPool.h"
 
 #ifndef SHORTLINK_SRC_HTTPSERVER_H_
 #define SHORTLINK_SRC_HTTPSERVER_H_
 
 namespace shortlink {
-
+enum ServerStatus { kRunning, kStopped };
+typedef boost::shared_ptr<boost::asio::ip::tcp::socket> SocketPointer;
 // Handles the connection from all the clients.
 // HttpServer will establish the ListenerThreadPool during the construction.
 // Example:
-//    std::unique_ptr<HttpServer> httpServer = make_unique<HttpServer>();
+//    std::unique_ptr<HttpServer> httpServer = std::make_unique<HttpServer>();
 //    // start the server
 //    httpServer->Start();
-public
+
 class HttpServer {
  public:
   // Constructor
-  HttpServer();
+  HttpServer(int threadPoolSize, boost::asio::io_context& context, int port);
+  // Delete default constructor
+  HttpServer() = delete;
   // Delete copy constructors
-  HttpServer(const HttpServer &) = delete;
-  HttpServer &operator=(const HttpServer &) = delete;
+  HttpServer(const HttpServer&) = delete;
+  HttpServer& operator=(const HttpServer&) = delete;
   // Destructor
   ~HttpServer();
-  // Start the http server
+  // Start the server
   void Start();
-  // Close the http server
+  // Close all the threads and async-works
   void Shutdown();
   // Get the server status
-  int GetServerStatus();
-  // Get the current number of clients
+  ServerStatus GetServerStatus() { return serverStatus; }
+  // Get the number of current serving clients
   int GetClientCount();
```

Comment:
if the implement will not change the member in class, use const member function to get function

Reply:
This is a reasonable review.

Reply:
const keyword added.